Compare-NET-Objects/VerifyConfig.cs

```diff
@@ -37,16 +37,15 @@ public void VerifySpec(ComparisonConfig config)
                     string msg = string.Format(
                         "Collection Matching Spec Type {0} should be a class, not a List.  Expected something like Customer, not List<Customer>.  See https://github.com/GregFinzer/Compare-Net-Objects/wiki/Comparing-Lists-of-Different-Lengths",
                         kvp.Key.Name);
-                    throw new Exception(msg);
+                    throw new ArgumentException(msg, nameof(config));

```

Comment:
Change type of an exception to provide caller more information about exception. I think that using raw Exception is not that good in most cases. We should use the specific one if it is possible.